core/src/main/java/com/datastrato/gravitino/storage/relational/service/CatalogMetaService.java

```diff
@@ -0,0 +1,196 @@
+    String metalakeName = identifier.namespace().level(0);
+    String catalogName = identifier.name();
+    Long metalakeId =
+        SessionUtils.getWithoutCommit(
+            MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeIdMetaByName(metalakeName));
+    if (metalakeId == null) {
+      throw new NoSuchEntityException(identifier.namespace().toString());
+    }
+    CatalogPO catalogPO =
+        SessionUtils.getWithoutCommit(
+            CatalogMetaMapper.class,
+            mapper -> mapper.selectCatalogMetaByMetalakeIdAndName(metalakeId, catalogName));
+    if (catalogPO == null) {
+      throw new NoSuchEntityException(identifier.toString());
+    }
+    return POConverters.fromCatalogPO(catalogPO, identifier.namespace());
+  }
+
+  public List<CatalogEntity> listCatalogsByNamespace(Namespace namespace) {
+    checkCatalogNamespace(namespace);
+    String metalakeName = namespace.level(0);
+    Long metalakeId =
+        SessionUtils.getWithoutCommit(
+            MetalakeMetaMapper.class, mapper -> mapper.selectMetalakeIdMetaByName(metalakeName));
+    if (metalakeId == null) {
+      throw new NoSuchEntityException(namespace.toString());
+    }
+    List<CatalogPO> catalogPOS =
+        SessionUtils.getWithoutCommit(
+            CatalogMetaMapper.class, mapper -> mapper.listCatalogPOsByMetalakeId(metalakeId));
+    return POConverters.fromCatalogPOs(catalogPOS, namespace);
+  }
+
+  public void insertCatalog(CatalogEntity catalogEntity, boolean overwrite) {
+    try {
+      checkCatalogNamespaceWithIdentifier(catalogEntity.nameIdentifier());
+      Long metalakeId =
+          SessionUtils.getWithoutCommit(
+              MetalakeMetaMapper.class,
+              mapper -> mapper.selectMetalakeIdMetaByName(catalogEntity.namespace().level(0)));
+      if (metalakeId == null) {
+        throw new NoSuchEntityException(catalogEntity.namespace().toString());
+      }
+      SessionUtils.doWithCommit(
+          CatalogMetaMapper.class,
+          mapper -> {
+            CatalogPO po = POConverters.initializeCatalogPOWithVersion(catalogEntity, metalakeId);
+            if (overwrite) {
+              mapper.insertCatalogMetaOnDuplicateKeyUpdate(po);
+            } else {
+              mapper.insertCatalogMeta(po);
+            }
+          });
+    } catch (RuntimeException re) {
+      if (re.getCause() != null
+          && re.getCause().getCause() != null
+          && re.getCause().getCause() instanceof SQLIntegrityConstraintViolationException) {
+        // TODO We should make more fine-grained exception judgments
+        // Usually throwing `SQLIntegrityConstraintViolationException` means that
+        // SQL violates the constraints of `primary key` and `unique key`.
+        // We simply think that the entity already exists at this time.
+        throw new EntityAlreadyExistsException(
+            String.format(
+                "Catalog entity: %s already exists", catalogEntity.nameIdentifier().name()));
```

Comment:
We need to use `catalogEntity.nameIdentifier()` as the name is not unique.

Reply:
Done